src/eos-google-chrome-installer

```diff
@@ -0,0 +1,218 @@
+
+        dbus_glib.DBusGMainLoop(set_as_default=True)
+        bus = dbus.SessionBus()
+        bus.add_signal_receiver(handler, dbus_interface="org.gnome.Shell.AppStore",
+                                signal_name="ApplicationsChanged")
+
+        loop.run()
+
+    def _set_as_default_browser(self):
+        mimetypes = ["text/html",
+                     "x-scheme-handler/http",
+                     "x-scheme-handler/https",
+                     "x-scheme-handler/about"]
+        for mimetype in mimetypes:
+            try:
+                subprocess.Popen(["xdg-mime", "default", "google-chrome.desktop", mimetype])
+            except OSError:
+                exit_with_error("Couldn't start xdg-mime. Are xdg-utils installed?")
+        try:
+            subprocess.Popen(["xdg-settings", "set", "default-web-browser", "google-chrome.desktop"])
+        except OSError:
+            exit_with_error("Couldn't start xdg-settings. Are xdg-utils installed?")
+
+    def _touch_done_file(self):
+        chrome_setup_done_path = pathlib.Path(os.path.expanduser("~/.config/google-chrome-initial-setup-done"))
+        if not chrome_setup_done_path.exists():
+            chrome_setup_done_path.touch()
+
+    def _post_install_chrome(self):
+        self._set_as_default_browser()
+        self._touch_done_file()
+
+    def _run_app_center_for_chrome(self):
+        # We use the APP ID as the one for GNOME Software, to let it choose the best one.
+        chrome_app_center_id = self.FLATPAK_CHROME_APP_ID
+
+        # FIXME: Ideally, we should be able to pass 'com.google.Chrome' to GNOME Software
+        # and it would do the right thing by opening the page for the app's branch matching
+        # the default branch for the apps' source remote. Unfortunately, this is not the case
+        # at the moment and fixing it is non-trivial, so we'll construct the full unique ID
+        # that GNOME Software expects, right from here, based on the remote's metadata.
+        default_branch = None
+        try:
+            remote = self._installation.get_remote_by_name(self.FLATPAK_REMOTE_EOS_APPS)
+        except GLib.Error as e:
+            logging.warning("Could not determine default branch for remote %s", self.FLATPAK_REMOTE_EOS_APPS)
+
+        default_branch = remote.get_default_branch()
+        if default_branch:
+            chrome_app_center_id = 'system/flatpak/{}/desktop/{}.desktop/{}'.format(self.FLATPAK_REMOTE_EOS_APPS,
+                                                                                    self.FLATPAK_CHROME_APP_ID,
+                                                                                    default_branch)
+        try:
+            subprocess.Popen(['gnome-software', '--install', chrome_app_center_id, '--interaction', 'none'])
+        except OSError as e:
+            exit_with_error("Could not launch Chrome: {}".format(repr(e)))
+
+        self._wait_for_installation()
```

Comment:
I think it would be nice to perform some extra checks between the wait finished and you actually claim to have "Chrome successfully installed", so that you don't clear the GSetting and do other post install tasks unless you are fairly sure that Chrome has been properly installed.